### PR TITLE
Transaction Middlewares & ErrorHandling

### PR DESCRIPTION
**Pull request by RodPaDev**
# Description

In this Pull Request, I've added middlewares & error handling, data-type validation happens automatically via postgreSQL, any further validation should be communicated to me

## Type of change

- [x] New feature (non-breaking change which adds functionality))
- [x] This change requires a documentation update

## Change status

- [x] Complete, tested, ready to review and merge

# How Has This Been Tested?

I thoroughly tested the edge cases in postman

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation, if relevant
- [x] I have run the app using my feature and ensured that no functionality is broken
- [x] My changes generate no new warnings
- [x] There are no merge conflicts

### CODE REVIEW

File: api/server.js [hunk 1]
```diff
@@ -32,4 +31,8 @@ server.get('/', (req, res) => {
   res.json('just cook it');
 });
 
+server.get('/api/docs', (req, res) => {
```
Comment (CarnunMP) [inline api/server.js]: Nice!

File: rest-api/recipe/models/recipe-models.js [hunk 2]
```diff
@@ -208,6 +210,7 @@ async function addRecipeTransaction(body) {
           const new_recipe_ingredients = ingredients.map(
             (ingredient, index) => {
               return {
+                // check this
```
Comment (CarnunMP) [inline rest-api/recipe/models/recipe-models.js]: Yeah... this seems wrong. We don't want each entry in the `new_recipe_ingredients` array to contain all of the ingredients in the req body, surely? :/

Comment (RodPaDev) [reply rest-api/recipe/models/recipe-models.js]: I will have a look at it today

Comment (RodPaDev) [reply rest-api/recipe/models/recipe-models.js]: Done for Fixed

Comment (CarnunMP) [reply rest-api/recipe/models/recipe-models.js]: Sweet!

**Pull request merged**